scipy/sparse/csgraph/tests/test_disjoint_set.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+from numpy.testing import assert_array_equal
+from scipy.sparse.csgraph import DisjointSet
+
+
+def test_linear_union_sequence():
+    n = 10
+    dis = DisjointSet(n)
+    assert dis.nc == n
+
+    for i in range(n - 1):
+        assert dis.union(i, i + 1)
+        assert dis.nc == n - 1 - i
+
+    roots = [dis.find(i) for i in range(n)]
+    assert_array_equal(0, roots)
+    assert not dis.union(0, n - 1)
+
+
+def test_self_unions():
+    n = 10
+    dis = DisjointSet(n)
+
+    for i in range(n):
+        assert not dis.union(i, i)
+    assert dis.nc == n
+
+    roots = [dis.find(i) for i in range(n)]
+    assert_array_equal(np.arange(n), roots)
+
+
+def test_equal_size_ordering():
+    n = 10
+    dis = DisjointSet(n)
+
+    for i in range(0, n, 2):
+        assert dis.union(i, i + 1)
```

Comment:
it feels like you should be passing `(i + 1, i)` if you want to test a and b get swapped.

Reply:
Thanks. I have increased the size of the set and used randomized ordering of `i` and `i + 1` so that we get a mixture of input orderings.